src/Coordination/SnapshotableHashTable.h

```diff
@@ -39,7 +98,12 @@ class SnapshotableHashTable
     /// Allows to avoid additional copies in updateValue function
     size_t current_version{0};
     size_t snapshot_up_to_version{0};
-    ArenaWithFreeLists arena;
+
+    /// Arena used for keys
+    /// we don't use std::string because it uses 24 bytes (because of SSO)
+    /// we want to always allocate the key on heap and use StringRef to it
+    GlobalArena arena;
```

Comment:
I am curious that why we used `ArenaWithFreeLists` before.

Now we use GlobalArena, but why not `Common/Allocator.h`. It looks having same features but traced by `MemoryTracker`.

Reply:
All our new/delete calls are tracked with `MemoryTracker`.
As this is a simple new/delete call I didn't see a reason to use something more complex like `Allocator.h`

> I am curious that why we used ArenaWithFreeLists before.

I'm not sure what were the benefits but it had huge downsides IMO.
`ArenaWithFreeLists` doesn't work for global usage like this because it doesn't deallocate memory on free, just adds it to free list. E.g. you create 50m nodes and delete all of them, all memory allocated for their paths will be kept in memory for no reason.
`ArenaWithFreeList` has bins with 2^n size. The overhead is much worse for larger bins and we end up with a lot of unused memory. In case of global allocator we will use jemallocs bins which are much closer to the real size we want to allocate (https://jemalloc.net/jemalloc.3.html#size_classes)